crates/relayer/src/chain/cosmos.rs

```diff
@@ -721,6 +727,33 @@ impl CosmosSdkChain {
         .await
     }
 
+    async fn do_maybe_register_counterparty_payee(
+        &self,
+        channel_id: &ChannelId,
+        port_id: &PortId,
+        counterparty_payee: &Signer,
+    ) -> Result<(), Error> {
+        let address = self.get_signer()?;
+        let key_pair = self.key()?;
+
+        // Take a write lock on the current account,
+        // excluding anybody from sending txs until we are done.
+        let mut opt_account = self.account.write().await;
```

Comment:
We take the lock here for the remaining of the function, ie. until the counterparty payee has been registered.